xarray/backends/netCDF4_.py

```diff
@@ -251,6 +251,12 @@ def get_dimensions(self):
         return FrozenOrderedDict((k, len(v))
                                  for k, v in iteritems(self.ds.dimensions))
 
+    def get_encoding(self):
+        encoding = {}
+        encoding['unlimited_dims'] = set(
+            [k for k, v in self.ds.dimensions.items() if v.isunlimited()])
```

Comment:
you can use a set comprehension here, e.g.,

    encoding['unlimited_dims'] = {k for k, v in self.ds.dimensions.items()
                                  if v.isunlimited()}